File: MyProjects/MAP/Assignment 7/Interpreter/Model/Statements/PrintStmt.cs
using System;
using System.Collections.Generic;
using System.Text;
using Interpreter.Model.Expressions;

namespace Interpreter.Model.Statements
{
    public class PrintStmt:IStmt
    {

        private IExpr expr;

        public PrintStmt(IExpr expression)
        {
            expr = expression;
        }

        public ProgramState execute(ProgramState state)
        {
            int val = expr.evaluate(state.SymbolTable);
            state.Output.add(val);
            return state;
        }

        public override string ToString()
        {
            return "print(" + expr + ")";
        }
    }
}

File: MyProjects/MAP/Practical Exam/Repeat Until Stmt/src/Model/Statements/PrintStmt.java
package Model.Statements;

import Model.Expressions.*;
import Model.ProgramState;
import Model.Utilities.*;

public class PrintStmt implements IStmt {
    private IExpr expr;

    public PrintStmt(IExpr expression){
        expr = expression;
    }

    @Override
    public ProgramState execute(ProgramState state){
        IList<Integer> list = state.getList();
        IDictionary<String, Integer> dic = state.getSymTable();
        IDictionary<Integer, Integer> heap=state.getHeap();
        int val = expr.evaluate(dic,heap);
        list.add(val);
        return null;
    }

    public String toString(){
        return "print(" + expr + ")";
    }
}
